Purchases/Public/Purchases.swift

```diff
@@ -1175,7 +1175,7 @@ public extension Purchases {
      */
     @available(iOS 12.2, macOS 10.14.4, macCatalyst 13.0, tvOS 12.2, watchOS 6.2, *)
     @objc(paymentDiscountForProductDiscount:product:completion:)
-    func paymentDiscount(forProductDiscount discount: SKProductDiscount,
+    func paymentDiscount(forProductDiscount discount: StoreProductDiscount,
```

Comment:
[Not needed for this PR]
We're going to either deprecate or fully remove the version of the method that takes in a `SKProductDiscount` soon, since this is something that we can do internally without asking customers to do another API call. 

https://app.shortcut.com/revenuecat/story/12136/public-api-deprecate-paymentdiscount-forproductdiscount-product-completion


Perhaps, in order to provide a heads up for devs about the change, we can keep (or reintroduce, since it'll probably be done as a separate PR) the old version of the method, which we'll deprecate as part of the task I linked to above, and then rename this one and make it internal. 

Thoughts? 

Also, @NachoSoto I saw your comment on the task, have you already started on this one?

Reply:
I haven't, I was planning to today.

Reply:
I changed these parameters back to `SKProductDiscount` when deprecated, so you can ignore it for now :)

Reply:
@NachoSoto Wait... so am I good with the merge conflicts that I did? Or do I need to fix something else? 😇

Reply:
Nope, you're good!